### PR TITLE
[Feature] New job posted notification

### LINKED ISSUE #9621: ❗ [Notification] New job posted
## ✨ Purpose

Let existing users know that a new job has been posted on the platform.

## ❗ Trigger

A pool is published.

## 👀 Audience

All users with the Applicant role (so all users).

### Email templates (English and French)

Hey {{user.firstname}}, 

You signed up to be notified when new opportunities are posted on GC Digital Talent.  

A new opportunity is now available! Find out if this is a fit for you and apply: 

((opportunity.title)hyperlinked) 

**standard email notification footer**

Objet :
Un nouveau poste est affiché sur Talents numériques du GC 

Corps du texte :
Vous avez accepté de recevoir des avis lorsqu’un nouveau poste est affiché sur Talents numériques du GC. 

Un nouveau poste y est maintenant affiché! À vous de juger s’il vous convient et de présenter votre candidature :
((opportunity.title)hyperlien)

**standard email notification footer**

### In-app string (English and French)
A new opportunity is now available! Find out if this is a fit for you and apply.

Un nouveau poste y est maintenant affiché! À vous de juger s’il vous convient et de présenter votre candidature.

### Link

Links to the job poster page

## 💾 Content

- This belongs to the "Job alerts" family of notifications (see #9555)
- Must store pool id?

## ✅ Acceptance Criteria

- [ ] Email template exists in GC Notify
- [ ] Notification class exists in Laravel
- [ ] Notification is triggered by appropriate event and send to appropriate users (including NOT any users who are ignoring this notification's Notification Family)
- [ ] Please confirm that the "Job alerts" notification family is not "checked" by default on the Notifications settings. This family really needs to be opt-in (not opt-out).
- [ ] Notification type added to schema
- [ ] Notification type rendered appropriately on frontend
- [ ] Tests
- [ ] Yes, we've added the in-app notification back in. Please ensure it's there.

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/9640
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10357
```

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #9621 

## 👋 Introduction

This branch adds a new notification, in-app and email, for users when a new pool is published.

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

1. Rebuild the app, using the "team" API key for GCNotify.
2. Log in as an admin, ensure your TBS email address is set for the user, make sure job alert notifications are enabled for in-app and email.
3. Publish a new pool

- [ ] received an email
- [ ] received an in-app notification

## 📸 Screenshot

image
image

## 🚚 Deployment

Add new environment variables for templates:
- GCNOTIFY_TEMPLATE_NEW_JOB_POSTED_EN
- GCNOTIFY_TEMPLATE_NEW_JOB_POSTED_FR

### CODE REVIEW

**Comment by petertgiles** [comment]
Could @gobyrne or @marc-donofrio please review the two new templates in GCNotify?

File: api/tests/Feature/Notifications/TriggerNewJobPostedTest.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            ->create([
+                'sub' => 'regularUser',
+                'ignored_email_notifications' => [],
+                'ignored_in_app_notifications' => [],
+            ]);
+    }
+
+    // no notification when an application is created or submitted
+    public function testNothingSentForDraftPool(): void
+    {
```
Comment (vd1992) [inline api/tests/Feature/Notifications/TriggerNewJobPostedTest.php:53]: Here

File: api/tests/Feature/Notifications/TriggerNewJobPostedTest.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            'fr' => 'new name',
+        ];
+        $pool->save();
+
+        Notification::assertNothingSent();
+    }
+
+    // triggers a notification when an application is moved into a final decision status
+    public function testNotifyWhenPublished(): void
+    {
```
Comment (vd1992) [inline api/tests/Feature/Notifications/TriggerNewJobPostedTest.php:71]: Here

File: api/tests/Feature/Notifications/TriggerNewJobPostedTest.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+        $pool->published_at = Carbon::now();
+        $pool->save();
+
+        Notification::assertSentTimes(NewJobPosted::class, 2);
+    }
+
+    // no notification when an application is closed or archived
+    public function testNothingSentForClosedPool(): void
+    {
```
Comment (vd1992) [inline api/tests/Feature/Notifications/TriggerNewJobPostedTest.php:87]: Here

**Review by vd1992 — COMMENTED** [review]
Some accidental copypasted comments, should be pool right not application

**Review by vd1992 — CHANGES_REQUESTED** [review]
Per the pictures I shared, the notifications aren't sending as I expect and I don't understand why/how

File: api/tests/Feature/Notifications/TriggerNewJobPostedTest.php (hunk 1 above)
Comment (petertgiles) [reply api/tests/Feature/Notifications/TriggerNewJobPostedTest.php]: Fixed in [fix test comments](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10358/commits/7cc36f9056382efd23c36ad65d7f0fec10ea8ce5)

**Comment by petertgiles** [comment]
Thanks for looking at this with me.  To document, if there's an uncaught exception from GCNotify it fails to send the in-app notification.  :grimacing:

**Review by vd1992 — APPROVED** [review]
🎉

**Comment by petertgiles** [comment]
> Thanks for looking at this with me. To document, if there's an uncaught exception from GCNotify it fails to send the in-app notification. 😬

I've seen [some discussions](https://github.com/laravel-notification-channels/channels/issues/16) online about this.  It seems like if you've got a synchronous notification you should be catching exceptions and firing a "notification failed" event but if you're using queuing the exception is fine because it causes the job to move about its state machine.  Since we're planning queuing very soon I think I'm going to leave it as-is instead of trying to figure out the event.

**Pull request merged**